Modules/Loadable/Segmentations/Widgets/qMRMLSegmentationGeometryWidget.cxx

```diff
@@ -0,0 +1,803 @@
+    vtkSegmentationConverter::GetReferenceImageGeometryParameterName(), geometryString );
+  qDebug() << Q_FUNC_INFO << "Reference image geometry of " << d->SegmentationNode->GetName()
+    << " has been set to '" << geometryString.c_str() << "'";
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidget::resampleLabelmapsInSegmentationNode()
+{
+  Q_D(qMRMLSegmentationGeometryWidget);
+  if (!d->SegmentationNode.GetPointer())
+    {
+    qCritical() << Q_FUNC_INFO << "No input segmentation specified";
+    return;
+    }
+  bool success = true;
+
+  // Check if master representation is binary or fractional labelmap (those are the only supported representations in segment editor)
+  std::string masterRepresentationName = d->SegmentationNode->GetSegmentation()->GetMasterRepresentationName();
+  if ( masterRepresentationName != vtkSegmentationConverter::GetBinaryLabelmapRepresentationName()
+    && masterRepresentationName != vtkSegmentationConverter::GetFractionalLabelmapRepresentationName() )
+    {
+    qCritical() << Q_FUNC_INFO << "Master representation needs to be a labelmap type, but '"
+      << masterRepresentationName.c_str() << "' found";
+    return;
+    }
+
+  std::vector< std::string > segmentIDs;
+  d->SegmentationNode->GetSegmentation()->GetSegmentIDs(segmentIDs);
+  for (std::vector< std::string >::const_iterator segmentIdIt = segmentIDs.begin(); segmentIdIt != segmentIDs.end(); ++segmentIdIt)
+    {
+    std::string currentSegmentID = *segmentIdIt;
+    vtkSegment* currentSegment = d->SegmentationNode->GetSegmentation()->GetSegment(*segmentIdIt);
+
+    // Get master labelmap from segment
+    vtkOrientedImageData* currentLabelmap = vtkOrientedImageData::SafeDownCast(
+      currentSegment->GetRepresentation(masterRepresentationName) );
+    if (!currentLabelmap)
+      {
+      qCritical() << Q_FUNC_INFO << "Failed to retrieve master representation from segment " << currentSegmentID.c_str();
+      continue;
+      }
+
+    // Resample
+    if (!vtkOrientedImageDataResample::ResampleOrientedImageToReferenceOrientedImage(currentLabelmap, d->GeometryImageData, currentLabelmap, false, true))
+      {
+      qCritical() << Q_FUNC_INFO << "Segment " << d->SegmentationNode->GetName() << "/" << currentSegmentID.c_str() << " failed to be resampled";
+      success = false;
+      continue;
+      }
+    }
+
+  if (success)
+    {
+    qDebug() << Q_FUNC_INFO << "Master representation '" << masterRepresentationName.c_str() << "' in each segment of segmentation "
+      << d->SegmentationNode->GetName() << " has been resampled to specified geometry";
+    d->SegmentationNode->Modified();
+    }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidget::geometryImageData(vtkOrientedImageData* outputGeometry)
+{
+  Q_D(qMRMLSegmentationGeometryWidget);
+  if (!outputGeometry)
+    {
+    return;
```

Comment:
should we set d->GeometryImageData to identity or something like that?

Reply:
This is a get function. I don't understand why we should reset it after getting it.